### ISSUE TITLE
aws_cdk_s3: enforce_ssl typo

### ISSUE THREAD

**Issue by poladinesh** [aws/aws-cdk#24915]
### Describe the issue

enforce_ssl parameter has a typo in python documentation under aws_cdk_s3 module

Currently the documentation says:

bucket = s3.Bucket(self, "Bucket",
    **enforce_sSL**=True
)

Correct Version:

bucket = s3.Bucket(self, "Bucket",
    **enforce_ssl**=True
)

### Links

https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_s3/README.html

**Comment by pahud** [comment]
Yes that should be fixed. While aws-cdk v1 [will reach end-of-support ](https://aws.amazon.com/blogs/developer/version-1-of-the-aws-cloud-development-kit-aws-cdk-is-now-in-maintenance-mode/)on June 1st I strongly recommend you check out the v2 document here instead https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_s3/Bucket.html

**Comment by poladinesh** [comment]
Hello @pahud,

Even the below links have references to **enforce_sSL** in few places under v2 version:

https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_s3/Bucket.html
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_s3/README.html

**Comment by stefanfreitag** [comment]
Hi, let me add the relevant line in the source code:

[https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-s3/lib/bucket.ts#L1617](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-s3/lib/bucket.ts#L1617)

```
 * new Bucket(scope, 'Bucket', {
 *   blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
 *   encryption: BucketEncryption.S3_MANAGED,
 *   enforceSSL: true,
 *   versioned: true,
 *   removalPolicy: RemovalPolicy.RETAIN,
 * });
```
The example in the source code seems to be correct. When transforming this example to Python code, keys change to snake_case: `removalPolicy` to `removal_policy` and also `blockPublicAccess` to `block_public_access`. For `enforceSSL`  it is `enforce_sSL` - unfortunately  not `enforce_ssl` as one would expect.

Not sure where this snake_case  magic happens though.
